### DISCUSSION TITLE
Add support for required web link arguments

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5532] [Feature Requests]
The button that is supposed to open a default mail client and open a draft to the Agent's agent is enabled even if there is no email specified

![Screenshot from 2020-02-07 14-15-22](https://user-images.githubusercontent.com/40512816/74062657-5b203600-49b4-11ea-951b-05c775ed9e2a.png)

In Chrome, the user just gets a warning regarding page update and then gets redirected to the same page
![Screenshot from 2020-02-07 14-15-28](https://user-images.githubusercontent.com/40512816/74062659-5b203600-49b4-11ea-9e11-2293d15eda9e.png)

I don't know what is the result in other browsers

Would it make sense to disable the button unless a valid email address is specified?

**Comment by maxpatiiuk** [comment]
I guess this applies to all weblinks.

There needs to be a way to specify a weblink argument as required, and if it is missing, the web link would be disabled

**Comment by grantfitzsimmons** [comment]
From @carlosmbe:

> **Is your feature request related to a problem? Please describe.**
> Having Specify able to prompt your mailing app is cool and all. But it would be better if it made a check that the field actually contained an email address and not some other combination of characters. IE: A phone number, a text message or a [foreign language (No Roman Characters, like Chinese, Korean or Japanese)](https://shkspr.mobi/blog/2016/09/why-cant-you-send-email-to-a-chinese-address/)
> 
> **Describe the solution you'd like**
> A check that it conforms to the standard [example@domain.com](mailto:example@domain.com) like [richardhendricks@piedpiper.com](mailto:richardhendricks@piedpiper.com)
> 
> **Describe alternatives you've considered**
> Just denying anything that is not part of ASCII. This would include Foreign Characters and emojis.
> 
> **This would prevent the screen shot below from being created.**
> <img width="1190" alt="Screenshot 2023-04-27 at 4 05 29 PM" src="https://user-images.githubusercontent.com/53784701/234996200-39ad8177-e770-473e-948b-2fda8e85d03f.png">
> <img width="885" alt="Screenshot 2023-04-27 at 4 05 39 PM" src="https://user-images.githubusercontent.com/53784701/234996201-484c3b47-6500-41ef-817a-ab24eee6b699.png">
>
